src/passes/OptimizeInstructions.cpp

```diff
@@ -2074,6 +2079,13 @@ struct OptimizeInstructions
     return areConsecutiveInputsEqualAndRemovable(left, right);
   }
 
+  // Similar to areConsecutiveInputsEqualAndFoldable, but only checks that they
+  // are equal (and not that they are foldable).
+  bool areConsecutiveInputsEqual(Expression* left, Expression* right) {
+    // TODO: optimize cases that must be equal but are *not* foldable.
```

Comment:
It would be helpful to provide an example of such a pattern here.

Reply:
I don't have one :smile: at least not a believable one. e.g. `(x++, x-1), (x--, x)` are equal (both return the original value of x) but we can't remove either one.